### PR TITLE
Implement all OpenAI job API status values

### PR DESCRIPTION
**Pull request by Ladicle**

### CODE REVIEW

File: dispatcher/internal/dispatcher/lifecycle_manager.go [hunk 1]
```diff
@@ -129,10 +131,28 @@ func (s *LifecycleManager) Reconcile(
 		}
 		if !expired {
 			requeueAfter := time.Until(expirationTime)
+			log.V(2).Info("Job is cancelled but not expired yet", "requeue-after", requeueAfter)
 			return ctrl.Result{Requeue: true, RequeueAfter: requeueAfter}, nil
 		}
+		log.Info("Delete the cancelled and expired job")
```
Comment (kenji-cloudnatix) [inline dispatcher/internal/dispatcher/lifecycle_manager.go:137]: Slightly prefer `Deleting`

**Review by kenji-cloudnatix — APPROVED** [review]
Thanks! LG!

File: dispatcher/internal/dispatcher/lifecycle_manager.go [hunk 2]
```diff
@@ -129,10 +131,28 @@ func (s *LifecycleManager) Reconcile(
...
+
+	if job.Status.Failed > 0 {
+		if err := s.store.UpdateJobState(jobData.JobID, jobData.Version, store.JobStatusFailed); err != nil {
+			log.Error(err, "Failed to update job state")
+			return ctrl.Result{}, err
+		}
+		var message string
+		for _, cond := range job.Status.Conditions {
+			if cond.Type == batchv1.JobFailed {
+				message = fmt.Sprintf("%s: %s", cond.Reason, cond.Message)
```
Comment (kenji-cloudnatix) [inline dispatcher/internal/dispatcher/lifecycle_manager.go:149]: No need to do in this PR, but I guess we can populate the `error` field.

https://platform.openai.com/docs/api-reference/fine-tuning/object

`finished_at` is another field we can populate.

Comment (Ladicle) [reply dispatcher/internal/dispatcher/lifecycle_manager.go:149]: Yes, I plan to implement it in other PRs.

**Pull request merged**